build_resolvers/lib/src/resolver.dart

```diff
@@ -254,17 +254,19 @@ class AnalyzerResolver implements ReleasableResolver {
   ///
   /// This includes the main library and existing part files.
   Future<List<ErrorsResult>> _syntacticErrorsFor(LibraryElement element) async {
-    final existingElements = [
-      element,
-      for (final part in element.parts)
-        // The source may be null if the part doesn't exist. That's not
-        // important for us since we only care about syntax
-        if (part.source.exists()) part,
-    ];
+    final existingSources = [element.source];
+
+    for (final part in element.parts2) {
+      var uri = part.uri;
+      // The source may be null if the part doesn't exist. That's not
```

Comment:
No, if it is `DirectiveUriWithSource`, then the `.source` is not `null`.

This cycle could be `parts2.map((e) => e.uri).whereType<DirectiveUriWithSource>().map((e) => e.source)`, and used with `...` in the list literal.

Reply:
I updated the comment a bit from the old language, I think its getting a bit much to try and jam into a list literal though.